static_frame/core/index.py

```diff
@@ -430,6 +430,13 @@ def __init__(self,
         # NOTE: automatic discovery is possible, but not yet implemented
         self._loc_is_iloc = loc_is_iloc
 
+    def __reversed__(self):
+        '''
+        Returns a reverse iterator on the index labels.
+        '''
+        for label in self.values[::-1]:
```

Comment:
I ran some time trials, and it looks like simply returning a reverse iterator of the underlying array is more than twice as fast to iterate over (it's a bit slower to call initially, but that difference is insignificant when compared to iteration time). So this should be fine:

```suggestion
        return reversed(self.values)
```